Backend/src/routes/siteRoutes.js

```diff
@@ -38,4 +38,20 @@ router.post('/', tokenChecker, async (req, res) => {
     }
 });
 
+router.patch('/:id', tokenChecker, async (req, res) => {
+    if (!req.body) {
+        return res.status(400).json(createError('Richiesta non valida', 400, 'Devi fornire le informazioni nel corpo della richiesta.'));
+    }
+
+    try {
+        const site = await service.updateSite(req.body, req.params.id);
+        res.status(204).json(site);
+    } catch (error) {
+        if (error.code === 400) {
+            return res.status(400).json(error);
+        }
+        res.status(500).json(error);
```

Comment:
Invece dell'if metti solo così
```suggestion
        res.status(error.code).json(error);
```